[Bug] [dinky-core] Click to trigger savepoint error

### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar issues.


### What happened

点击 “Savapoint 触发” 报错。

![A6B0DB96-AC24-4C72-9890-888550649CEB](https://github.com/user-attachments/assets/154a96fd-b579-4463-be25-38252870cef4)




```log
2025-01-01 10:45:53 Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
2025-01-01 10:45:53     at java.lang.StringLatin1.charAt(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.lang.String.charAt(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.net.HostPortrange.<init>(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.net.URLPermission$Authority.<init>(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.net.URLPermission.parseURI(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.net.URLPermission.init(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at java.net.URLPermission.<init>(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at sun.net.www.protocol.http.HttpURLConnection.URLtoSocketPermission(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(Unknown Source) ~[?:?]
2025-01-01 10:45:53     at cn.hutool.http.HttpConnection.getOutputStream(HttpConnection.java:461) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpRequest.sendFormUrlEncoded(HttpRequest.java:1367) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpRequest.send(HttpRequest.java:1342) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpRequest.doExecute(HttpRequest.java:1188) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpRequest.execute(HttpRequest.java:1051) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpRequest.execute(HttpRequest.java:1027) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at cn.hutool.http.HttpUtil.post(HttpUtil.java:230) ~[hutool-all-5.8.31.jar:5.8.31]
2025-01-01 10:45:53     at org.dinky.api.FlinkAPI.post(FlinkAPI.java:119) ~[dinky-core-1.2.0.jar:?]
2025-01-01 10:45:53     at org.dinky.api.FlinkAPI.triggerSavePoint(FlinkAPI.java:189) ~[dinky-core-1.2.0.jar:?]
2025-01-01 10:45:53     at org.dinky.api.FlinkAPI.savepoints(FlinkAPI.java:178) ~[dinky-core-1.2.0.jar:?]
2025-01-01 10:45:53     at org.dinky.job.JobManager.savepoint(JobManager.java:384) ~[dinky-core-1.2.0.jar:?]
2025-01-01 10:45:53     at org.dinky.service.impl.TaskServiceImpl.savepointTaskJob(TaskServiceImpl.java:480) ~[dinky-admin-1.2.0.jar:?]
```

我想“智能停止” 也会报这个错。

Bug in there（not jobmanger host in url）:

```diff
diff --git a/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java b/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java
index 912f3f0dd..4e7fcf059 100644
--- a/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java
+++ b/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java
@@ -112,7 +112,7 @@ public class FlinkAPI {
     }
 
     private JsonNode post(String route, String body) {
-        String url = NetConstant.SLASH + route;
+        String url = address + NetConstant.SLASH + route;
         if (!address.startsWith(NetConstant.HTTP) && !address.startsWith(NetConstant.HTTPS)) {
             url = NetConstant.HTTP + url;
         }
```

### What you expected to happen

可以带 savepoint 停止任务。

### How to reproduce

start job, and click "Savapoint 触发"

### Anything else



### Version

1.2.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java
+++ b/dinky-core/src/main/java/org/dinky/api/FlinkAPI.java
@@ -112,7 +112,7 @@ public class FlinkAPI {
     }
 
     private JsonNode post(String route, String body) {
-        String url = NetConstant.SLASH + route;
+        String url = address + NetConstant.SLASH + route;
         if (!address.startsWith(NetConstant.HTTP) && !address.startsWith(NetConstant.HTTPS)) {
             url = NetConstant.HTTP + url;
         }

